### ISSUE TITLE
Configurable LR Schedulers

### ISSUE THREAD

**Issue by radiradev** [ecmwf/anemoi-core#653]
### Is your feature request related to a problem? Please describe.

Currently the learning rate scheduler is hard-coded. Similar to https://github.com/ecmwf/anemoi-core/pull/588 we should have arbitrary pytorch learning rate schedulers available to users. 

### Describe the solution you'd like

Instantiate schedulers directly, for instance as in [here](https://github.com/ashleve/lightning-hydra-template/blob/bddbc24b82ab6ccfa6243e815a49dc5bfe8d4144/configs/model/mnist.yaml#L9-L14): 

```
scheduler:
  _target_: torch.optim.lr_scheduler.ReduceLROnPlateau
  _partial_: true
  mode: min
  factor: 0.1
  patience: 10
```
